mk-repl-modules/MkReplModules.hs

```diff
@@ -8,10 +8,21 @@ import System.FilePath
 main :: IO ()
 main = do
   cwd <- getCurrentDirectory
-  let rootDir = takeDirectory cwd
+  rootDir <- findRootDir cwd
+  putStrLn $ "Root dir: " ++ rootDir
   mkReplModules (rootDir </> "base-compat")           ["Repl"]
   mkReplModules (rootDir </> "base-compat-batteries") ["Repl", "Batteries"]
 
+-- Go up until there is 'cabal.project'
```

Comment:
Thanks for doing this—this is something I had put on my todo list but didn't bother finishing. :)